test/test_zip_map.cpp

```diff
@@ -0,0 +1,274 @@
+
+// Copyright (c) 2022 Tristan Brindle (tcbrindle at gmail dot com)
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include "catch.hpp"
+
+#include <array>
+#include <iostream>
+#include <list>
+
+#include "test_utils.hpp"
+
+namespace {
+
+constexpr bool test_zip_map()
+{
+    {
+        int arr1[] = { 0, 1, 2, 3, 4, 5, 6, 7, 8, 9 };
+        double arr2[] = {0, 100, 200, 300, 400};
+
+        auto zipped = flux::zip_map(
+            [](int first, double second) {
+                return first + static_cast<int>(second);
+            },
+            flux::mut_ref(arr1), flux::mut_ref(arr2)
+        );
+
+        using Z = decltype(zipped);
+
+        static_assert(flux::sequence<Z>);
+        static_assert(flux::bidirectional_sequence<Z>);
+        static_assert(flux::random_access_sequence<Z>);
+        static_assert(flux::sized_sequence<Z>);
+        static_assert(flux::bounded_sequence<Z>);
+
+        static_assert(flux::sequence<Z const>);
+        static_assert(flux::bidirectional_sequence<Z const>);
+        static_assert(flux::random_access_sequence<Z const>);
+        static_assert(flux::sized_sequence<Z const>);
+        static_assert(flux::bounded_sequence<Z const>);
+
+        static_assert(std::same_as<flux::element_t<Z>, int>);
+        static_assert(std::same_as<flux::value_t<Z>, int>);
+        static_assert(std::same_as<flux::rvalue_element_t<Z>, int>);
+
+        // Believe it or not these are correct -- we've zipped two ref adaptors
```

Comment:
No change required, but this comment has slipped past copy-and-paste -- there's nothing surprising about the following lines with `zip_map`